src/core/server/saved_objects/migrationsv2/model.ts

```diff
@@ -516,14 +546,57 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
       throwBadResponse(stateP, res);
     }
   } else if (stateP.controlState === 'REINDEX_SOURCE_TO_TEMP_INDEX') {
+    // TINA We used to transform and index documents in one action.
+    // This is not split into two steps: One for transforms, the next for the bulk index
+    // We follow a similar control flow as for
+    // outdated document search -> outdated document transform -> transform documents bulk index
+    // collecting issues along the way rather than failing
+    // REINDEX_SOURCE_TO_TEMP_INDEX handles the document transforms
+    const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
+    if (Either.isRight(res)) {
+      if (stateP.corruptDocumentIds.length === 0 && stateP.transformErrors.length === 0) {
+        return {
+          ...stateP,
+          controlState: 'REINDEX_SOURCE_TO_TEMP_INDEX_BULK', // handles the actual bulk indexing into temp index
+          transformedDocs: [...res.right.processedDocs],
+        };
+      } else {
+        // we don't have any transform issues with the current batch of outdated docs BUT
+        // we have carried through previous transformation issues.
+        // The migration will ultimately fail BUT
+        // search throught remaining docs for more issues and pass the previous failures along on state
+        return {
+          ...stateP,
+          controlState: 'REINDEX_SOURCE_TO_TEMP_READ',
+        };
+      }
+    } else {
+      // we have failures from the current batch of documents and add them to the lists
+      const left = res.left;
+      if (isLeftTypeof(left, 'documents_transform_failed')) {
+        return {
+          ...stateP,
+          controlState: 'REINDEX_SOURCE_TO_TEMP_READ',
+          corruptDocumentIds: [...stateP.corruptDocumentIds, ...left.corruptDocumentIds],
+          transformErrors: [...stateP.transformErrors, ...left.transformErrors],
+        };
+      } else {
+        // should never happen
+        throwBadResponse(stateP, res as never);
+      }
+    }
+  } else if (stateP.controlState === 'REINDEX_SOURCE_TO_TEMP_INDEX_BULK') {
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
     if (Either.isRight(res)) {
       return {
         ...stateP,
         controlState: 'REINDEX_SOURCE_TO_TEMP_READ',
+        // we're still on the happy path with no transformation failures seen.
+        corruptDocumentIds: [],
+        transformErrors: [],
       };
     } else {
-      const left = res.left;
+      const left = Either.left;
```

Comment:
Splitting out the indexing from the transformation step resulted in a slight change in the return type we observe in the model.

Reply:
btw it shows that we don't have tests for this branch

Reply:
You're both right. 

I actually added this back in at the [last minute](https://github.com/elastic/kibana/pull/96986/commits/c909b3a2c81c21c680b2c835b47f5ba72590a781) without considering the response type here is identical to the response type from `stateP.controlState === 'TRANSFORMED_DOCUMENTS_BULK_INDEX'`. Both `REINDEX_SOURCE_TO_TEMP_INDEX_BULK` & `TRANSFORMED_DOCUMENTS_BULK_INDEX` issue the same `Action.bulkOverwriteTransformedDocuments` and the model should treat the response identically.

I've removed the last minute addition.

Reply:
I added this code back in at the [last minute](https://github.com/elastic/kibana/pull/96986/commits/c909b3a2c81c21c680b2c835b47f5ba72590a781) not considering that the model should treat the response from `Actions.bulkOverwriteTransformedDocuments` in the same way as for `controlState === TRANSFORMED_DOCUMENTS_BULK_INDEX`.

I removed it again.